source/_posts/2018-03-14-GoogleCast.md

```diff
@@ -0,0 +1,401 @@
+
+Now that we have an established connection and our player is aware that a connection is established, we can play a MediaItem on our receiving device.
+
+To do so we need to cover the following steps:
+
+- create GCKMediaMetadata
+- create GCKMediaInformation
+- start video on cast
+
+In our CastManager add the following functions that together help create a GCKMediaInformation, containing GCKMediaMetadata, which will be send to the receiving device.
+
+```
+func buildMediaInformation(with title: String, with description: String, with studio: String, with duration: TimeInterval, with movieUrl: String, with streamType: GCKMediaStreamType, with thumbnailUrl: String?) -> GCKMediaInformation {
+    let metadata = buildMetadata(with: title, with: description, with: studio, with: thumbnailUrl)
+
+    let mediaInfo = GCKMediaInformation.init(contentID: movieUrl, streamType: streamType, contentType: "video/m3u8", metadata: metadata, streamDuration: duration, mediaTracks: nil, textTrackStyle: nil, customData: nil)
+
+    return mediaInfo
+}
+
+private func buildMetadata(with title: String, with description: String, with studio: String, with thumbnailUrl: String?) -> GCKMediaMetadata {
+    let metadata = GCKMediaMetadata.init(metadataType: .movie)
+    metadata.setString(title, forKey: kGCKMetadataKeyTitle)
+    metadata.setString(description, forKey: "description")
+    
+    let deviceName = sessionManager.currentCastSession?.device.friendlyName ?? studio
+    metadata.setString(deviceName, forKey: kGCKMetadataKeyStudio)
+    
+    if let thumbnailUrl = thumbnailUrl, let url = URL(string: thumbnailUrl) {
+        metadata.addImage(GCKImage.init(url: url, width: 720, height: 480))
+    }
+
+    return metadata
+}
+```
+
+Now that we can create a GCKMediaInformation we need to create a function that will allow us to start a video on our connected receiver device.
+
+For this add the following function in your CastManager:
+
+```
+func startSelectedItemRemotely(_ mediaInfo: GCKMediaInformation, at time: TimeInterval, completion: (Bool) -> Void) {
+    let castSession = sessionManager.currentCastSession
+
+    if castSession != nil {
+        let options = GCKMediaLoadOptions()
+        options.playPosition = time
+        options.autoplay = true
+        castSession?.remoteMediaClient?.loadMedia(mediaInfo, with: options)
+        completion(true)
+
+        sessionStatus = .alreadyConnected
+    } else {
+        completion(false)
+    }
+}
+```
+
+This will take the GCKMediaInformation we will create with the previous created functions and, if a cast session is available, load media on the remote client. Once this happens the completion will return true a cast session exists and false if we can not send a the GCKMediaInformation because there is no active cast session at the moment.
+
+
+Now that we have all the componesnts together to start streaming our first video go ahead and update your `listenForCastConnection` function in the Player class.
```

Comment:
components

And do you need the "together"

Now that we have alle the components needed to start streaming our first video....